scripts/kind-with-registry.sh

```diff
@@ -42,37 +56,138 @@ if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true
+    if ! az account show > /dev/null 2>&1; then
+        echo "Please login to Azure CLI using 'az login'"
+        exit 1
+    fi
 
-  if [[ -z "${SERVICE_ACCOUNT_SIGNING_KEY}" ]]; then
-    echo "'SERVICE_ACCOUNT_SIGNING_KEY' is not set."
-    exit 1
+    if [ -z "${AZWI_RESOURCE_GROUP}" ]; then
+      echo "AZWI_RESOURCE_GROUP environment variable required - Azure resource group to store required Workload Identity artifacts"
+      exit 1
+    fi
+    if [ -z "${AZWI_LOCATION}" ]; then
+      echo "AZWI_LOCATION environment variable required - Azure location where required Workload Identity artifacts reside"
+      exit 1
+    fi
+    if [ "$(az group exists --name "${AZWI_RESOURCE_GROUP}" --output tsv)" == 'false' ]; then
+      echo "Creating resource group '${AZWI_RESOURCE_GROUP}' in '${AZWI_LOCATION}'"
+      az group create --name "${AZWI_RESOURCE_GROUP}" --location "${AZWI_LOCATION}" --output none --only-show-errors --tags creationTimestamp="${TIMESTAMP}" jobName="${JOB_NAME}" buildProvenance="${BUILD_PROVENANCE}"
+    fi
+    if ! az storage account show --name "${AZWI_STORAGE_ACCOUNT}" --resource-group "${AZWI_RESOURCE_GROUP}" > /dev/null 2>&1; then
+      echo "Creating storage account '${AZWI_STORAGE_ACCOUNT}' in '${AZWI_RESOURCE_GROUP}'"
+      az storage account create --resource-group "${AZWI_RESOURCE_GROUP}" --name "${AZWI_STORAGE_ACCOUNT}" --allow-blob-public-access true --output none --only-show-errors --tags creationTimestamp="${TIMESTAMP}" jobName="${JOB_NAME}" buildProvenance="${BUILD_PROVENANCE}"
+    fi
+    # Ensure that we our connection to storage is inherited from the existing Azure login context
+    unset AZURE_STORAGE_KEY
+    unset AZURE_STORAGE_ACCOUNT
+    if ! az storage container show --name "${AZWI_STORAGE_CONTAINER}" --account-name "${AZWI_STORAGE_ACCOUNT}" > /dev/null 2>&1; then
+      echo "Creating storage container '${AZWI_STORAGE_CONTAINER}' in '${AZWI_STORAGE_ACCOUNT}'"
+      az storage container create --name "${AZWI_STORAGE_CONTAINER}" --account-name "${AZWI_STORAGE_ACCOUNT}" --public-access blob --output none --only-show-errors
+    fi
+    export SERVICE_ACCOUNT_ISSUER="https://${AZWI_STORAGE_ACCOUNT}.blob.core.windows.net/${AZWI_STORAGE_CONTAINER}/"
+    AZWI_OPENID_CONFIG_FILEPATH="${REPO_ROOT}/openid-configuration.json"
+    cat <<EOF > "${AZWI_OPENID_CONFIG_FILEPATH}"
+{
+  "issuer": "${SERVICE_ACCOUNT_ISSUER}",
+  "jwks_uri": "${SERVICE_ACCOUNT_ISSUER}openid/v1/jwks",
+  "response_types_supported": [
+    "id_token"
+  ],
+  "subject_types_supported": [
+    "public"
+  ],
+  "id_token_signing_alg_values_supported": [
+    "RS256"
+  ]
+}
+EOF
+  openssl genrsa -out "${SERVICE_ACCOUNT_SIGNING_KEY_FILEPATH}" 2048
+  openssl rsa -in "${SERVICE_ACCOUNT_SIGNING_KEY_FILEPATH}" -pubout -out "${SERVICE_ACCOUNT_SIGNING_PUB_FILEPATH}"
+  AZWI_JWKS_JSON_FILEPATH="${REPO_ROOT}/jwks.json"
+  "${AZWI}" jwks --public-keys "${SERVICE_ACCOUNT_SIGNING_PUB_FILEPATH}" --output-file "${AZWI_JWKS_JSON_FILEPATH}"
```

Comment:
This will not be required if we use the keys generated as part of the cluster creation. We might be able to drop dependency on `azwi` CLI.